shared/macros-oval.jinja

```diff
@@ -115,15 +118,28 @@
 
 {{#
   Macro to check if a parameter in a configuration file is set (Object definition).
-  This function can take five parameters:
+  This macro can take six parameters:
     - path (String): Path to the configuration file to be checked.
+    - section (String): If set, the parameter will be checked only within the given section defined by [section].
     - prefix_regex (String): Regular expression to be used in the beginning of the OVAL text file content check.
     - parameter (String): The parameter to be checked in the configuration file.
     - separator_regex (String): Regular expression to be used as the separator of parameter and value in a configuration file. If spaces are allowed, this should be included in the regular expression.
     - missing_parameter_pass (boolean): If set, the check will also pass if the parameter is not present in the configuration file (default is applied).
 #}}
-{{%- macro oval_line_in_file_object(path='', prefix_regex='^[\s]*', parameter='', separator_regex='[\s]+', missing_parameter_pass=false) -%}}
+{{%- macro oval_line_in_file_object(path='', section='', prefix_regex='^[\s]*', parameter='', separator_regex='[\s]+', missing_parameter_pass=false) -%}}
 {{%- set suffix_id = "" -%}}
+{{%- if section %}}
+{{%- if missing_parameter_pass %}}
+{{#
+  There is no need for having a regular expression with a capture
+  group "(\S*)" when checking if the parameter is not present in
+  the configuration file.
+#}}
+{{%- set regex = "^\s*\["+section+"\].*(?:\\n\s*[^[\s].*)*\\n"+prefix_regex+parameter+separator_regex+"\S*\s*(?:|(?:#.*))?$" -%}}
```

Comment:
Can you make improve the `regex` variable construction? Trying to reuse as much the regular expressions. I might help decreasing maintenance costs.

Reply:
Is the trailing part `"\S*\s*(?:|(?:#.*))?$"` of the regex needed?

Reply:
@yuumasato great point, it might not be needed, I'll check it

Reply:
@ggbecker @yuumasato I have tried with regex101 and I think it's not needed in situation we only want to check the option is not present. I have removed it in 4918c60.

Reply:
Great. It looks way better now.